v1/ao/internal/reporter/reporter_grpc.go

```diff
@@ -122,6 +117,20 @@ type grpcReporter struct {
 	insecureSkipVerify bool // for testing only: if true, skip verifying TLS cert hostname
 }
 
+// verifyServiceKey verifies if the service key is valid
```

Comment:
This function only checks if the service key contains 2 parts (`key:name`). But it doesn't verify that the key is valid (e.g. no check for lenth == 64 or that it only contains hex characters). Maybe change function name to something like `verifyServiceKeyFormat` or add more verification.